pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -232,15 +240,27 @@ private void processNewSegment(SegmentMetadata segmentMetadata, URI finalSegment
             .info("Moved segment {} from temp location {} to {}", segmentName, currentSegmentLocation.getAbsolutePath(),
                 finalSegmentLocationURI.getPath());
       } catch (Exception e) {
+        // Cleanup the Zk entry and the segment from the permanent directory if it exists.
         LOGGER
             .error("Could not move segment {} from table {} to permanent directory", segmentName, tableNameWithType, e);
+        _pinotHelixResourceManager.deleteSegment(tableNameWithType, segmentName);
```

Comment:
Add a log message to show that the new uploaded segment is being deleted.

Reply:
Done.